src/town_cmd.cpp

```diff
@@ -249,7 +249,7 @@ static TownDrawTileProc * const _town_draw_tile_procs[1] = {
  */
 static inline DiagDirection RandomDiagDir()
 {
-	return (DiagDirection)(3 & Random());
+	return (DiagDirection)(Random() & (DIAGDIR_END - 1));
```

Comment:
I know the code is technically correct... but the trickery with `DIAGDIR_END` and then doing a bitmask tingles my spider senses. What if DIAGDIR_END would become 6 (because we use hexagonal tiles). Then this code becomes horribly broken. However, just using modulo might not be what you really want to express.
You actually want to say: `return (DiagDirection)RandomRange(DIAGDIR_END);` because you just want a random number that is in the range of valid DiagDirection values.